client/verta/verta/dataset/_dataset.py

```diff
@@ -62,3 +68,77 @@ def _path_component_to_repr_lines(path_component_msg):
             lines.append("    SHA-256 checksum: {}".format(path_component_msg.sha256))
 
         return lines
+
+    def _set_commit_and_blob_path(self, commit, blob_path):
+        """
+        Associate this blob with a commit and path to enable downloads.
+
+        Parameters
+        ----------
+        commit : :class:`verta._repository.commit.Commit`
+            Commit this blob was gotten from.
+        blob_path : str
+            Location of this blob within its Repository.
+
+        """
+        self._commit = commit
+        self._blob_path = blob_path
+
+    def download(self, component_path, filepath=None, chunk_size=32*(10**3)):
+        """
+        Downloads `component_path` from this dataset if ModelDB-managed versioning was enabled.
+
+        Parameters
+        ----------
+        component_path : str
+            Original path of the file in this dataset to download.
+        filepath : str, optional
+            Filepath to download `component_path` to. If not provided, the file will be downloaded
+            to the current directory.
+        chunk_size : int, default 32 kB
+            Number of bytes to download at a time.
+
+        """
+        if self._commit is None and self._blob_path is None:
+            raise RuntimeError(
+                "this dataset cannot be used for downloads;"
+                " consider using `commit.get()` to obtain a download-capable dataset"
+                " if ModelDB-managed versioning was enabled"
+            )
+        if filepath is None:
+            filepath = os.path.basename(component_path)
+
+        # TODO: remove this check when the os.rename() step is fixed to sidestep collisions
+        if os.path.exists(filepath):
+            raise OSError("{} already exists".format(filepath))
```

Comment:
This is the contrary of what we had agreed, right?

Reply:
You're correct. Updated.